CIP-XXXX/README.md

```diff
@@ -0,0 +1,120 @@
+In particular, combining those two wrappers avoided needing to migrate the node's on-disk files to a new schema.
+But since this CIP requires a change to the schema, it's also an opportunity to eliminate that special case: if a schema change is already happening, the marginal overhead of nested wrappers for Byron no longer justifies the special case.
+
+None of these tags --- even the original Byron block tag --- have ever influenced hashes, so it is possible for their schema to change; it's simply never happened before.
+
+## Follow-Up Change: Transactions
+
+Removing the HFC wrapper from transactions is an improvement in two ways.
+- Just as today's HFC tags leak implementation details into the codecs, today's `cardano-cli` command-line interface leaks those same implementation details by forcing the user to pick the HFC tag for their transaction.
+  (This is not apparent as a burden for users today, since there's essentially only one CLI interface for Cardano.)
+- As explained in the "Follow-Up Change: Backwards-Compatibility Conversions" bullet above, the node doesn't actually need an HFC wrapper for incoming transactions: the current state of the mempool determines which protocol version to use when decoding the bytes of the incoming transaction.
+
+If the Ledger Team does accidentally leave ambiguity among the codecs of two adjacent protocol versions, then there would be a risk of misinterpreting a submitted transaction near a protocol version boundary.
+Even today, however, that (unlikely) risk instead motivates a dedicated new field specifying the intended protocol versions that is within the scope of the transaction's signature to prevent the adversary from tampering---the HFC wrapper itself is the wrong approach to mitigating that risk, since it's not signed.
+
+# Path to Active
+
+## Acceptance Criteria
+
+- The latest release of all Cardano node implementations:
+    - switches on the protocol version instead of switching on some abstraction of the protocol version,
+    - stores annotations alongside their persisted blocks with that same protocol version instead of some abstraction thereof,
+    - and disallows HFC wrappers for submitted transactions.
+- All nodes also maintain the older codecs for backwards-compatibility for at least one year (TODO what duration to require here?).
+
+## Implementation Plan
+
+Each node implementation team would implement this CIP in their own node.
+Because all implementations currently constrain the single integer negotiatied as of the node's handshake, every implementation would need to deploy this change in a release at approximately the same time, which could be difficult to arrange seamlessly.
+(TODO has this challenge already been explicitly called out anywhere?
+Perhaps we should branch the negotated version so that it "knows" about different implementations?
+That doesn't sound sustainable either...)
```

Comment:
> Because all implementations currently constrain the single integer negotiatied as of the node's handshake, every implementation would need to deploy this change in a release at approximately the same time, which could be difficult to arrange seamlessly.

I don't understand this, isn't the entire idea of the N2N protocol version that nodes with different maximum supported N2N versions can coexist seamlessly (until we remove old N2N versions after some hard fork)? Or is this what you mean by "at approximately the same time"?

Reply:
I pushed up a slight clarification. What do you think now?